src/compiler/dataStructures.ts

```diff
@@ -0,0 +1,768 @@
+// Map
+namespace ts {
+    export interface MapCommon<K, V> {
+        /** Removes all entries. */
+        clear(): void;
+        /** Removes the entry with the given key. */
+        delete(key: K): void;
+        /** Gets the value associated with a key. */
+        get(key: K): V | undefined;
+        /** Whether any entry in the map has the given key. */
+        has(key: K): boolean;
+        /** Associate a value with a key. */
+        set(key: K, value: V): void;
+    }
+
+    /** String-keyed Map. */
+    export interface SMap<V> extends MapCommon<string, V> {

```

Comment:
`NumberKeyedMap`, `StringKeyedMap`, `ObjectBackedMap`. With good autcomplete, there's no real reason for confusing or ambiguous hungarianesque notation.